mount/temp_unix.go

```diff
@@ -40,23 +41,20 @@ func SetTempMountLocation(root string) error {
 
 // CleanupTempMounts all temp mounts and remove the directories
 func CleanupTempMounts(flags int) (warnings []error, err error) {
-	mounts, err := Self()
+	mounts, err := mountinfo.GetMounts(mountinfo.PrefixFilter(tempMountLocation))
```

Comment:
Alas we can't have a "fast path" I talked about earlier in here, since `tempMountLocation` is not a mount point per se.

In case containerd itself is doing all those mounts, those can be saved in a list, and we should go over that list in here (instead of reading mountinfo) -- that would be a worthwhile optimization.

In case a user does those mounts, this is unfortunate and there's nothing we can do but read mountinfo.

In any case, this can be addressed later.